app/controllers/users/notes_controller.rb

```diff
@@ -4,6 +4,8 @@ class NotesController < ApplicationController
     expose_decorated :notes, :user_notes
     expose :user, find_by: :username
```

Comment:
Let's create a new controller — `BaseController`.
All controllers that need authorization will inherit this controller. In our case, it is a `Users::NotesController`.

BaseController will look like this:
```ruby
class BaseController < ApplicationController
  include Authorization

  before_action :authenticate_user!
  before_action :authorize_resource!

  protect_from_forgery with: :exception
end
```
Each controller which inherits from the `BaseController` will implement `authorize_resource!` method. The `authenticate_user!` method is from the devise gem.

Note: We should remove the following code from `ApplicationController`.
```ruby
  include Authorization
  ...
  protect_from_forgery with: :exception
```

In this case, you will no need to check in the policy if the user logged in or not because he will be authenticated before authorization.
And Policy will check only if the user (who has definitely logged in) has the rights to do some action or not.